src/iterators/hybrid_reader.c

```diff
@@ -88,72 +86,62 @@ static void insertResultToHeap_Metric(HybridIterator *hr, RSIndexResult *child_r
   *upper_bound = worst->data.num.value;
 }
 
-static void insertResultToHeap_Aggregate(HybridIterator *hr, RSIndexResult *res, RSIndexResult *child_res,
+static void insertResultToHeap_Aggregate(HybridIterator *hr, RSIndexResult *child_res,
                                          RSIndexResult *vec_res, double *upper_bound) {
 
-  AggregateResult_AddChild(res, vec_res);
-  AggregateResult_AddChild(res, child_res);
-  RSIndexResult *hit = IndexResult_DeepCopy(res);
-  IndexResult_ResetAggregate(res); // Reset the current result.
-  ResultMetrics_Add(hit, hr->base.ownKey, RS_NumVal(vec_res->data.num.value));
+  RSIndexResult *res = NewHybridResult();
+  AggregateResult_AddChild(res, IndexResult_DeepCopy(vec_res));
```

Comment:
Why is a deep copy needed here now?

Reply:
Same as before (I switched from adding the children to the parent, then deep copying the parent and resetting it, to adding a deep copy of the children to a new result that will own the copies)
We deep-copy because if we reach here, we are in a scenario where we need the full results to calculate the text scores later